app/api/orders.py

```diff
@@ -302,7 +302,7 @@ def before_delete_object(self, order, view_kwargs):
             raise ConflictException({'source': ''}, 'You cannot delete a placed/completed paid order.')
 
     decorators = (jwt_required, api.has_permission(
-        'is_user_itself', methods="PATCH,DELETE", fetch="id", fetch_as="order_id", model=Order),)
+        'is_user_itself', methods="PATCH,DELETE", fetch="user_id", model=Order),)
```

Comment:
Why `user_id`?

Reply:
Because we need user_id. `is_user_itself` needs the user id. We need to check if this user is owner of the order. So we compare the current logged in user with the `user_id` of the order owner.

Reply:
But doesn't `before_update_object()` does that already?

Reply:
1. You added this decorator, so I thought it is needed. 
2. If this is redundant then we can remove this later or in another PR. This is super needed right now in the frontend.

Reply:
The code to add `id` key in `view_kwargs` requires having this decorator, that's why I added it. Yes, we can indeed fix redundancies in another PR. If the current code does the job for now, no issues with that :)